xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/document/DocumentUserManager.java

```diff
@@ -56,24 +59,47 @@ public class DocumentUserManager implements UserManager
     @Inject
     private Provider<XWikiContext> xwikiContextProvider;
 
+    @Inject
+    private WikiDescriptorManager wikiDescriptorManager;
+
     @Override
     public boolean exists(UserReference userReference)
     {
         boolean result;
 
-        // For the reference to point to an existing user it needs to satisfy 2 conditions:
+        // For the reference to point to an existing user it needs to satisfy 3 conditions:
+        // - the wiki of the document exists
         // - the document exists
         // - it contains an XWiki.XWikiUsers xobject
         XWikiContext xcontext = this.xwikiContextProvider.get();
         XWiki xwiki = xcontext.getWiki();
         DocumentReference userDocumentReference = ((DocumentUserReference) userReference).getReference();
+        String userWikiId = userDocumentReference.getWikiReference().getName();
+        try {
+            if (!this.wikiDescriptorManager.exists(userWikiId)) {
+                result = false;
+            } else {
+                result = exists(xcontext, xwiki, userDocumentReference);
+            }
+        } catch (WikiManagerException e) {
+            this.logger.warn("Failed to determine if wiki [{}] exists. Considering it's not the case. Cause: [{}]",
```

Comment:
I'm not a huge fan of this logic, but you did not really introduce this concept and unfortunately #exists was declared as not throwing any exception (which is what should happen here).

Reply:
I don't know much what could cause such exception in the wiki manager, but for me it makes sense to handle it like that here. Not sure exists should actually throw it: if you have a problem there, it probably means there would be other issues later if you tried to use the user. And it's probably not the job of the consumer of this API to handle the problem caused by the wiki manager. So all in all, it sounds like a good compromise to consider the user just doesn't exist in such case.

Reply:
otoh, in case of initialization issue you might wrongfully indicate to the caller that the user does not exist. In the context of `R140401000XWIKI15460` the consequence is the removal of all the user's notification preferences, which is pretty bad.
I think propagating the exception is the safest thing to do here.

Reply:
The whole point is that an API like this should not lie when it does not know the answer (and `false` is a lie here) because it's a very common cause of cache mess and that's really not the kind of problem you want to have when talking about users.